File: Exercise/LuoGu/Code_2023_7_30/P1321.java
package Code_2023_7_30;

import java.util.Scanner;

public class P1321 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String str = sc.nextLine();
        String boyStart = "", girlStart = "";
        int boyCnt = 0, girlCnt = 0;
        String BOY = "boy", GIRL = "girl";
        for (int i = 0; i < str.length(); i++) {
            char ch = str.charAt(i);
            if (ch != '.') {
                if (BOY.contains("" + ch)) {
                    girlStart = "";
                    if (BOY.contains(boyStart + ch)) {
                        if (boyStart.length() == 0) {
                            boyCnt++;
                        }
                        boyStart += ch;
                    } else {
                        boyCnt++;
                        boyStart = "" + ch;
                    }
                } else {
                    boyStart = "";
                    if (GIRL.contains(girlStart + ch)) {
                        if (girlStart.length() == 0) {
                            girlCnt++;
                        }
                        girlStart += ch;
                    } else {
                        girlCnt++;
                        girlStart = "" + ch;
                    }
                }
            }
        }
        System.out.printf("%d\n%d\n", boyCnt, girlCnt);
        sc.close();
    }
}

File: Exercise/LuoGu/Code_2023_7_30/P1321.cpp
#include <iostream>
#include <algorithm>
#include <math.h>
#include <vector>
#include <string>

using namespace std;

int main(void)
{

    string str;
    cin >> str;
    int boy_cnt = 0, girl_cnt = 0;
    string boy_start = "", girl_start = "";
    string BOY = "boy", GIRL = "girl";
    for (char i : str)
    {
        if (i != '.')
        {
            if (BOY.find(i) != string::npos)
            {
                girl_start = "";
                if (BOY.find(boy_start + i) != string::npos)
                {
                    if (boy_start == "")
                    {
                        boy_cnt++;
                    }
                    boy_start += i;
                }
                else
                {
                    boy_cnt++;
                    boy_start = i;
                }
            }
            else
            {
                boy_start = "";
                if (GIRL.find(girl_start + i) != string::npos)
                {
                    if (girl_start == "")
                    {
                        girl_cnt++;
                    }
                    girl_start += i;
                }
                else
                {
                    girl_cnt++;
                    girl_start = i;
                }
            }
        }
    }
    cout << boy_cnt << endl
         << girl_cnt << endl;
    return 0;
}